Foundation.hs

```diff
@@ -130,14 +130,16 @@ instance YesodAuth App where
     -- Override the above two destinations when a Referer: header is present
     redirectToReferer _ = True
 
-    getAuthId creds = runDB $ do
-        muser <- getBy $ UniqueUser (credsPlugin creds) (credsIdent creds)
+    getAuthId Creds{..} = runDB $ do
```

Comment:
I'm going to extract this into its own module and test it before making changes. This is a pretty critical path.